### ISSUE TITLE
unable to compile 1.1.0-pre2 on rhel6 based systems

### ISSUE THREAD

**Issue by sbagmeijer** [openssl/openssl#551]
Hello,

We are trying to update from the Alpha 1 version to the Alpha 2 version but seems something changed in the Alpha 2 that makes it not compile any more on rhel6 based linux versions.

It affects both i386 and x64, Alpha 1 however still works fine. (rhel 7 works fine also).

This is the gcc we use
gcc -v
```
Using built-in specs.
Target: i686-redhat-linux
Configured with: ../configure --prefix=/usr --mandir=/usr/share/man --infodir=/usr/share/info --with-bugurl=http://bugzilla.redhat.com/bugzilla --enable-bootstrap --enable-shared --enable-threads=posix --enable-checking=release --with-system-zlib --enable-__cxa_atexit --disable-libunwind-exceptions --enable-gnu-unique-object --enable-languages=c,c++,objc,obj-c++,java,fortran,ada --enable-java-awt=gtk --disable-dssi --with-java-home=/usr/lib/jvm/java-1.5.0-gcj-1.5.0.0/jre --enable-libgcj-multifile --enable-java-maintainer-mode --with-ecj-jar=/usr/share/java/eclipse-ecj.jar --disable-libjava-multilib --with-ppl --with-cloog --with-tune=generic --with-arch=i686 --build=i686-redhat-linux
Thread model: posix
gcc version 4.4.7 20120313 (Red Hat 4.4.7-16) (GCC)
```

It breaks on this:
```
gcc -I.. -I../.. -I../modes -I../include -I../../include  -fPIC -DOPENSSL_PIC -DOPENSSL_THREADS -pthread -DDSO_DLFCN -DHAVE_DLFCN_H -Wa,--noexecstack -march=i486 -DL_ENDIAN -Wall -O3 -fomit-frame-pointer -DOPENSSL_BN_ASM_PART_WORDS -DOPENSSL_IA32_SSE2 -DOPENSSL_BN_ASM_MONT -DOPENSSL_BN_ASM_GF2m -DSHA1_ASM -DSHA256_ASM -DSHA512_ASM -DMD5_ASM -DRMD160_ASM -DAES_ASM -DVPAES_ASM -DWHIRLPOOL_ASM -DGHASH_ASM -DECP_NISTZ256_ASM -c arch/async_null.c -o arch/async_null.o
ar  r ../../libcrypto.a async.o async_err.o arch/async_posix.o arch/async_win.o arch/async_null.o
/usr/bin/ranlib ../../libcrypto.a || echo Never mind.
make[2]: Leaving directory `/home/ulyaoth/BUILD/openssl-1.1.0-pre2/crypto/async'
if [ -n "libcrypto.so.1.1 libssl.so.1.1" ]; then \
                (cd ..; make libcrypto.so.1.1); \
        fi
Bareword found where operator expected at util/mkdef.pl line 1573, near "s/\./_/gr"
Unquoted string "r" may clash with future reserved word at util/mkdef.pl line 1573.
syntax error at util/mkdef.pl line 1573, near "s/\./_/gr"
Execution of util/mkdef.pl aborted due to compilation errors.
/usr/bin/ld:crypto.map:1: syntax error in VERSION script
```

```
We build it like this:
./config --openssldir=/usr/local/ulyaoth/ssl/openssl1.1.0 shared
make depend
make all
make rehash
make
```

Does anyone see something obvious that I could change easy to make it work?

**Comment by levitte** [comment]
What perl version are you using?  It seems that it doesn't understand the 'r' modifier...

**Comment by levitte** [comment]
You could try and see if this patch helps:

    -			my $baseversion = $1 =~ s/\./_/gr;
    +			(my $baseversion = $1) =~ s/\./_/g;

**Comment by sbagmeijer** [comment]
@levitte I will give that a try now thank you for the quick response!

Also Rhel 6 comes with the following perl version:
perl -v
```
This is perl, v5.10.1 (*) built for i386-linux-thread-multi
```

**Comment by levitte** [comment]
I'm interested to know what else might fail regarding perl...

**Comment by sbagmeijer** [comment]
Well that fixed it :+1: 

/usr/local/ulyaoth/ssl/openssl1.1.0/bin/openssl version
OpenSSL 1.1.0-pre2 (alpha) 14 Jan 2016

**Comment by levitte** [comment]
Great!  Thanks for trying it out.

**Comment by sbagmeijer** [comment]
@levitte no worries if you want me to try something else to make sure the perl works let me know.

I really appreciate the quick response now I can release the 1.1.0 Alpha 2 rpm this evening :).

**Comment by kroeckx** [comment]
This seems to be fixed in 56afc18714664e8fb395a2e143e48be3ab146d43, so closing.
